scripts/run.py

```diff
@@ -0,0 +1,487 @@
+                if not interactive:
+                    failed = True
+                else:
+                    if confirm("%s failed. Abort?" % name):
+                        abort("Aborting at user request.")
+                    else:
+                        failed = True
+        return failed
+
+    def load_project_info(self, path):
+        info_path = os.path.join(path, "project.csv")
+        if not os.path.exists(info_path):
+            create_project_info(path)
+        with open(info_path) as csv_file:
+            reader = csv.reader(csv_file)
+            names = next(reader)
+            data = next(reader)
+            info = dict(zip(names, data))
+            self.info("[%s] Load project info" % info["name"])
+            return info
+
+    def get_report(self, project_path, kind):
+        report_path = os.path.join(project_path, self.config.get(kind))
+        for root, subdir, files in os.walk(project_path):
+            for f in files:
+                if f == self.config.get(kind):
+                    with open(os.path.join(root, f)) as report:
+                        return report.read()
+
+        return None
+
+    def write_report(self, content, project_path, kind):
+        report_path = os.path.join(project_path, self.config.get(kind))
+        with open(report_path, 'w') as report:
+            return report.write(content)
+
+    def info(self, msg):
+        if self.config.get("debug_info"):
+            self.logger.log(msg)
+
+    def error(self, msg):
+        if self.config.get("debug_info"):
+            self.logger.error(msg)
+
+    def checkout_latest_tag(self, project_name, project_path):
+        self.info("[Import][%s] Checkout latest tag..." % project_name)
+        self.local("git fetch --tags --depth 1", project_path)
+        self.local_canfail("Load latest tag", "latestTag=$( git describe --tags `git rev-list --tags --max-count=1` )", directory=project_path)
+        self.local("git checkout $latestTag", project_path)
+
+    def has_file_with_ending(self, folder, extension):
+        for root, dirs, files in os.walk(folder):
+            for f in files:
+                if f.endswith(extension):
+                    return True
+
+    def get_build_systems(self, project_path):
+        systems = []
+        if self.has_file_with_ending(project_path, "pom.xml"):
+            systems.append("maven")
+        if self.has_file_with_ending(project_path, ".sbt"):
+            systems.append("sbt")
+        if self.has_file_with_ending(project_path, "gradle"):
+            systems.append("gradle")
+        if self.has_file_with_ending(project_path, "Vagrantfile"):
+            systems.append("vagrant")
+        return systems
+
+    def is_build_system_supported(self, src_path):
+        project_build_systems = self.get_build_systems(src_path)
+        return any(system in self.config.get("supported_build_systems") for system in project_build_systems)
+
+class Pipeline:
```

Comment:
What is the point of this class?

Reply:
Do you actually need the `Pipeline*` classes at all? In which case would you need more than one instance?